pages/pipelines/group_step.md.erb

```diff
@@ -0,0 +1,78 @@
+# Group Step
+ 
+A *group* step groups up parallel jobs into logical groups and allows displaying them on the build page in the Buildkite UI.  
+
+{:toc}
+ 
+You can scale your build pipelines in Buildkite to support an unlimited number of jobs. However, a large number of jobs on the build page can look messy. If pipelines contain 4-5 steps that can be logically grouped together, you can put them in a group using the `group` step type.  
+ 
+Job groups allow creating and then displaying logical groups in the Buildkite UI. They also help manage dependencies between a collection of steps in a dag (for example, step X depends on everything in a group Y).
```

Comment:
The DAG stuff is also unrelated afaik, it's only mentioned because DAG + Groups isn't 100% tested yet. 

We should ask the team if there is an exhaustive list of what you can use in a group step? `notify`, `env` and `depends` at least?

Reply:
On my TODO list for tonight.